src/amo/components/InstallButtonWrapper/index.js

```diff
@@ -0,0 +1,141 @@
+/* @flow */
+import makeClassName from 'classnames';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { compose } from 'redux';
+
+import GetFirefoxButton from 'amo/components/GetFirefoxButton';
+import AMInstallButton from 'core/components/AMInstallButton';
+import { UNKNOWN } from 'core/constants';
+import { withInstallHelpers } from 'core/installAddon';
+import { getVersionById } from 'core/reducers/versions';
+import { getClientCompatibility, isFirefox } from 'core/utils/compatibility';
+import type { GetFirefoxButtonTypeType } from 'amo/components/GetFirefoxButton';
+import type { WithInstallHelpersInjectedProps } from 'core/installAddon';
+import type { UserAgentInfoType } from 'core/reducers/api';
+import type { InstalledAddon } from 'core/reducers/installations';
+import type { AddonVersionType } from 'core/reducers/versions';
+import type { AddonType } from 'core/types/addons';
+import type { AppState } from 'disco/store';
+
+import './styles.scss';
+
+export type Props = {|
+  _getClientCompatibility?: typeof getClientCompatibility,
+  addon: AddonType,
+  className?: string,
+  defaultButtonText?: string,
+  defaultInstallSource: string,
+  getFirefoxButtonType: GetFirefoxButtonTypeType,
+  puffy?: boolean,
+|};
+
+type InternalProps = {|
+  ...Props,
+  ...WithInstallHelpersInjectedProps,
+  clientApp: string,
+  currentVersion: AddonVersionType | null,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+  userAgentInfo: UserAgentInfoType,
+|};
+
+export const InstallButtonWrapperBase = (props: InternalProps) => {
+  const {
+    _getClientCompatibility = getClientCompatibility,
+    addon,
+    className,
+    clientApp,
+    currentVersion,
+    defaultButtonText,
+    defaultInstallSource,
+    enable,
+    getFirefoxButtonType,
+    hasAddonManager,
+    isAddonEnabled,
+    install,
+    installStatus,
+    installTheme,
+    setCurrentStatus,
+    puffy,
+    uninstall,
+    userAgentInfo,
+  } = props;
+
+  let isCompatible = false;
+  let compatibility;
+
+  if (addon && isFirefox({ userAgentInfo })) {
```

Comment:
There are a couple of reasons I did it this way:

1. If we know we're not on Firefox then we don't need to call `getClientCompatibility` at all. 
2. If we call `getClientCompatibility` all we get back is a reason that we're not compatible, so we'd have to look specifically for `INCOMPATIBLE_NOT_FIREFOX` as the reason to know it's not Firefox. I'd rather encapsulate the logic for that in `compatibility.js` (via `isFirefox`), rather than have that specific logic in here.

Do you think those reasons justify doing it the way I've done it?

Reply:
yes 👍  I think that justifies this.

Reply:
Quick question tho - should a test be added for this? You have a test that covers that `getClientCompatibility` is called, should we also have a test that it isn't called if it's not firefox and the default isCompatible value is used?

Reply:
Good point. I've added two tests for this.